arangod/RocksDBEngine/RocksDBCollection.cpp

```diff
@@ -714,7 +721,9 @@ Result RocksDBCollection::truncate(transaction::Methods& trx, OperationOptions&
 
   // avoid OOM error for truncate by committing earlier
   uint64_t const prvICC = state->options().intermediateCommitCount;
-  state->options().intermediateCommitCount = std::min<uint64_t>(prvICC, 10000);
+  auto const tmp = std::min<uint64_t>(prvICC, 10000);
+  state->options().intermediateCommitCount = tmp;
+  _statistics._numTruncate.count(tmp);
```

Comment:
This is not correct. `tmp` will contain the intermediate commit threshold value, which is not the number of documents removed/truncated.
The number of documents truncated is only determined further below in this method, and is only known in total when the `for` loop is completed.